File: backend/shared/validators.py
"""Validadores compartilhados entre features."""
import re

CNPJ_PATTERN = re.compile(r"^\d{14}$")
CHAVE_ACESSO_PATTERN = re.compile(r"^\d{44}$")
CHAVE_ACESSO_NACIONAL_PATTERN = re.compile(r"^\d{50}$")


def validar_cnpj(cnpj: str) -> bool:
    return bool(CNPJ_PATTERN.match(cnpj))


def validar_chave_acesso(chave: str) -> bool:
    """Valida formato da chave de acesso (44 dígitos municipal ou 50 dígitos nacional).

    Args:
        chave: String da chave de acesso.

    Returns:
        True se o formato é válido.
    """
    return bool(CHAVE_ACESSO_PATTERN.match(chave)) or bool(CHAVE_ACESSO_NACIONAL_PATTERN.match(chave))

File: backend/src/validators.ts
const CNPJ_PATTERN = /^\d{14}$/
const CHAVE_ACESSO_PATTERN = /^\d{44}$/
const CHAVE_ACESSO_NACIONAL_PATTERN = /^\d{50}$/

export function validarCnpj(cnpj: string): boolean {
  return CNPJ_PATTERN.test(cnpj)
}

export function validarChaveAcesso(chave: string): boolean {
  return CHAVE_ACESSO_PATTERN.test(chave) || CHAVE_ACESSO_NACIONAL_PATTERN.test(chave)
}
